pkg/cmd/image/list.go

```diff
@@ -188,50 +190,129 @@ type imagePrinter struct {
 	quiet, noTrunc, digestsFlag, namesFlag bool
 	tmpl                                   *template.Template
 	client                                 *containerd.Client
-	contentStore                           content.Store
+	provider                               content.Provider
 	snapshotter                            snapshots.Snapshotter
 }
 
-func (x *imagePrinter) printImage(ctx context.Context, img images.Image) error {
-	ociPlatforms, err := images.Platforms(ctx, x.contentStore, img.Target)
+type image struct {
+	blobSize int64
+	size     int64
+	platform platforms.Platform
+	config   *ocispec.Descriptor
+}
+
+var ReadBlob = containerdutil.ReadBlobWithCache()
```

Comment:
can we make this part of `imagePrinter` and pass it to `read/readManifest/readIndex` as an argument? I think that's better than global var?

Reply:
Mmm...

What about instead we instanciate ReadBlob cache inside containerdutil/helpers and use it just as a drop in replacement for content.ReadBlob?

The point is: ReadBlob is used solely for reading image manifest, index and config objects, which never change for a given digest.
There is no reason to ever invalidate the cache on this one.

Reply:
Doing that ^ with the latest push (also with your other comment fixed).
Let me know what you think (I will leave this comment unresolved for now).

Reply:
Pushed.

PTAL at your convenience.

Thanks @djdongjin !

Reply:
@djdongjin also adding another minor change: https://github.com/containerd/nerdctl/compare/1b401474ae01df7956eeb51c837c9b33f0d7ab55..80731ff39a21db7b9b58d4faebaf6ef8009a5cc8

Replacing testing media types with using existing containerd helpers for that.
Makes it slightly more future proof.